pkg/plugin/plugin.go

```diff
@@ -116,142 +136,119 @@ func getPluginConfig(rollout *v1alpha1.Rollout) (*GlooEdgeTrafficRouting, error)
 	return &glooplatformConfig, nil
 }
 
-func (r *RpcPlugin) getRouteTables(ctx context.Context, rollout *v1alpha1.Rollout, pluginConfig *GlooEdgeTrafficRouting) ([]*GlooMatchedRouteTable, error) {
-	if pluginConfig.RouteTableSelector == nil {
-		return nil, fmt.Errorf("routeTable selector is required")
-	}
+func getStableServiceName(rollout *v1alpha1.Rollout) string {
+	return rollout.Spec.Strategy.Canary.StableService
+}
 
-	if strings.EqualFold(pluginConfig.RouteTableSelector.Namespace, "") {
-		r.LogCtx.Debugf("defaulting routeTableSelector namespace to Rollout namespace %s for rollout %s", rollout.Namespace, rollout.Name)
-		pluginConfig.RouteTableSelector.Namespace = rollout.Namespace
-	}
+func getCanaryServiceName(rollout *v1alpha1.Rollout) string {
+	return rollout.Spec.Strategy.Canary.CanaryService
+}
 
-	var rts []*gloov1.RouteTable
+func (r *RpcPlugin) maybeConvertSingleToMulti(routeTables []routeTableWithDestinations) {
+	for i := range routeTables {
+		for j := range routeTables[i].Destinations {
+			if routeTables[i].Destinations[j].DestinationsParent.GetMulti() != nil {
```

Comment:
We need tho modify the original a bit later (here:[maybeConvertSingleToMulti](https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-glooedge/pull/1/files#diff-fc137d62147eb6a9c89c0566479ff219320ca691f2505631ebacbe513bcc8108R153)), I thought it might be confusing to mix both styles. I can change it if you think it's ok.